android/src/main/java/com/tailscale/ipn/MainActivity.kt

```diff
@@ -182,12 +186,38 @@ class MainActivity : ComponentActivity() {
     }
   }
 
-  private fun login(url: String) {
-    // (jonathan) TODO: This is functional, but the navigation doesn't quite work
-    // as expected.  There's probably a better built in way to do this.  This will
-    // unblock in dev for the time being though.
-    val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
-    startActivity(browserIntent)
+  private fun login(urlString: String) {
+    lifecycleScope.launch {
+      try {
+        Notifier.state.collect { state ->
```

Comment:
This block deserves a comment as to why it's here (i.e. we're waiting for login to complete and then relaunching the MainActivity.